server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessHealth.java

```diff
@@ -37,20 +44,55 @@ public class ClusterAwarenessHealth implements Writeable {
      */
     public ClusterAwarenessHealth(ClusterState clusterState, ClusterSettings clusterSettings, String awarenessAttributeName) {
         // This property will govern if we need to show unassigned shard info or not
-        boolean displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
-        clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
-            awarenessAttributeName,
-            displayUnassignedShardLevelInfo,
-            clusterState
-        );
+        boolean displayUnassignedShardLevelInfo;
+        ClusterAwarenessAttributesHealth clusterAwarenessAttributesHealth;
+        clusterAwarenessAttributesHealthMap = new HashMap<>();
+        // Helper function to check if we need health for all or for one awareness attribute.
+        boolean displayAllAwarenessAttribute = shouldDisplayAllAwarenessAttributeHealth(awarenessAttributeName);
+        if (!displayAllAwarenessAttribute) {
+            displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
+            clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
+                awarenessAttributeName,
+                displayUnassignedShardLevelInfo,
+                clusterState
+            );
+            clusterAwarenessAttributesHealthMap.put(awarenessAttributeName, clusterAwarenessAttributesHealth);
+        } else {
+            List<String> awarenessAttributeList = clusterSettings.get(
+                AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING
+            );
+            for (String awarenessAttribute : awarenessAttributeList) {
+                displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
+                clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
+                    awarenessAttribute,
+                    displayUnassignedShardLevelInfo,
+                    clusterState
+                );
+                clusterAwarenessAttributesHealthMap.put(awarenessAttribute, clusterAwarenessAttributesHealth);
+            }
+        }
```

Comment:
We can minimize the code duplication , by constructing `awarenessAttributeList` as per `displayAllAwarenessAttribute` .  

```
        List<String> awarenessAttributeList;
        if (displayAllAwarenessAttribute) {
            awarenessAttributeList = new ArrayList<>();
            awarenessAttributeList.add(awarenessAttributeName);
        } else {
            awarenessAttributeList = clusterSettings.get(
                AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING
            );
        }
        for (String awarenessAttribute : awarenessAttributeList) {
            displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
            clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
                awarenessAttribute,
                displayUnassignedShardLevelInfo,
                clusterState
            );
            clusterAwarenessAttributesHealthMap.put(awarenessAttribute, clusterAwarenessAttributesHealth);
        }
```